src/components/fields/image-upload/image-upload.tsx

```diff
@@ -211,17 +212,20 @@ export const ImageUploadInner = (props: IGenericFieldProps<IImageUploadSchema>)
 	);
 
 	const renderImageReviewModal = () => {
+		const maxRule = validation?.find((rule) => "max" in rule);
 		return (
 			<ImageReview
 				accepts={acceptedFileTypes}
 				dimensions={dimensions}
 				id={id}
 				className={className}
 				maxFiles={maxFiles}
+				maxFilesErrorMessage={maxRule?.errorMessage || ERROR_MESSAGES.UPLOAD("photo").MAX_FILES(maxFiles)}
```

Comment:
may I suggest to put the fallback in `ImageError`? because `maxFilesErrorMessage` is an optional prop there and it is clearer to see the is a default message there

in addition, do use the `MAX_FILES_WITH_REMAINING` error message instead because user should still be able to upload some more images in the review modal.
ref: https://github.com/LifeSG/web-frontend-engine/blob/main/src/components/fields/image-upload/image-input/image-input.tsx#L156